### PR TITLE
Add ability to build images using Docker client

### PR DESCRIPTION
**Pull request by whummer**
Add ability to build images using Docker client (both Cmd and Sdk client).

### CODE REVIEW

File: tests/integration/docker_utils/test_docker.py [hunk 1]
```diff
@@ -773,6 +782,37 @@ def test_run_container_automatic_pull(self, docker_client: ContainerClient):
         stdout, _ = docker_client.run_container("alpine", command=["echo", message], remove=True)
         assert message == stdout.decode(config.DEFAULT_ENCODING).strip()
 
+    @pytest.mark.skip_offline
+    @pytest.mark.parametrize("custom_context", [True, False])
+    @pytest.mark.parametrize("dockerfile_as_dir", [True, False])
+    def test_build_image(self, docker_client: ContainerClient, custom_context, dockerfile_as_dir):
+        dockerfile_dir = new_tmp_dir()
```
Comment (dfangl) [inline tests/integration/docker_utils/test_docker.py:789]: Maybe we can use pytest's tmp_path here, which returns a temporary pathlib path, which will automatically be removed after the test by pytest? this way we can make sure the path is removed after the test, while not having to rm_rf afterwards.

**Review by dfangl — APPROVED** [review]
LGTM (once the tests pass :)), just some minor nits regarding test temporary paths, not critical though :+1:

File: tests/integration/docker_utils/test_docker.py (hunk 1 above)
Comment (whummer) [reply tests/integration/docker_utils/test_docker.py]: Great point! 👍 updated..

Comment (dfangl) [reply tests/integration/docker_utils/test_docker.py]: Now the dockerfile_dir and the context_dir are always identical, I guess thats not supposed to be like it? 
tmp_path is a single location, we would have to use subfolders for that

Comment (whummer) [reply tests/integration/docker_utils/test_docker.py]: Outch - really sloppy on my part. Thanks for double-checking again! 👍

**Pull request merged**